### ISSUE TITLE
Grant Pricing Request Feature

### ISSUE THREAD

**Issue by calexh-sar** [sartography/cr-connect#5]
Similar to the GPR feature in the current CRC, we need a way for Coordinators to look up procedure pricing by CPT code or name, pick the procedure and add it to a cart.  Once the cart is complete with all the procedures needed for the study, the cart is download in spreadsheet format to be used to determine the overall cost of the study based on other factors like number of participants and number of times the procedure will be performed.

The current spreadsheet, which was uploaded in 2018 has 4868 rows and six columns...
- Procedure Description
- CPT Code
- EPIC CDM Code
- MC Price
- Study Price
- UPG Professional Fees

This spreadsheet is actually a consolidation of two spreadsheets, one from the Medical Center (MC) which has the cost of executing the procedure, e.g., how much did the actual X-Ray cost- with film, tech's time, etc. and the other from the University Physicians Group (UPG) which has the physician cost, e.g., how much it costs for the physician to review the X-Ray and write a report.  (In discuss now with Claire and Lydia about how the two spreadsheets are consolidated)  Each piece of the consolidate spreadsheet can be updated a different times.  Additionally, new procedures are added to the master spreadsheet between major updates.  Major updates are when all costs associated with each procedure are updated by either group, MC or UPG, or at the same time.

Although these major updates are infrequent, CRC2 needs to be able to be aware of when major updates are made.  If a study is in progress at the time of a major update, the following rules will apply:
- If a study has not started a cart, all new cart items will be at the updated prices.
- If a study has an existing cart, all current pricing will remain, but any new pricing will be a at the updated prices.
- If a study is on Hold, all current pricing will be update to the current pricing if Re-activated.
- If a study has pricing based on a version older than two updates prior (need to ask about this Edge Case)

**Comment by calexh-sar** [comment]
From 01/29 email from Shane and Lydia in which he wrote:

Here are the improvements we gathered. Let me know if you have any questions.

[https://app.zenhub.com/files/223421082/399e70e9-d30c-404e-ae2e-58584e9197ab/download](https://app.zenhub.com/files/223421082/399e70e9-d30c-404e-ae2e-58584e9197ab/download)

**Comment by calexh-sar** [comment]
Staying with currect UVA implemented app
